panda/src/navigation/navMeshNode.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file movieAudio.h
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+#ifndef NAVMESHNODE_H
+#define NAVMESHNODE_H
+
+#include <string>
+#include "geom.h"
+#include "pandaFramework.h"
+#include "pandaSystem.h"
+#include "navMesh.h"
+#include <string>
+
+class NavMeshNode: public PandaNode
+{
+PUBLISHED:
+  NavMeshNode(const std::string &name, PT(NavMesh) nav_mesh);
+private:
+  PT(NavMesh) _nav_mesh;
+
+public:
+  
+  ~NavMeshNode();
+};
```

Comment:
Class NavMeshNode needs a docstring

Since this class (indirectly) inherits from TypedObject, it needs TypeHandle boilerplate for use in Python (see the stuff at the bottom of MovieAudio and CollisionNode et al), this means adding this to the bottom of the class:

```cpp
public:
  static TypeHandle get_class_type() {
    return _type_handle;
  }
  static void init_type() {
    PandaNode::init_type();
    register_type(_type_handle, "NavMeshNode",
                  PandaNode::get_class_type());
  }
  virtual TypeHandle get_type() const {
    return get_class_type();
  }
  virtual TypeHandle force_init_type() {init_type(); return get_class_type();}

private:
  static TypeHandle _type_handle;
```

and this to the top of the .cxx file, below the includes:
```cpp
TypeHandle NavMeshNode::_type_handle;
```

And this to the config_navigation.cxx file, in the init function:
```
  NavMeshNode::init_type();
```